code/game/objects/items/weapons/twohanded.dm

```diff
@@ -608,3 +609,211 @@
 	if(!.)
 		return
 	toggle_item_bump_attack(user, FALSE)
+
+/// Chainsword & Chainsaw
+/obj/item/weapon/twohanded/chainsaw
+	name = "chainsaw"
+	desc = "A chainsaw. Good for turning big things into little things."
+	icon = 'icons/obj/items/weapons/misc.dmi'
+	worn_icon_list = list(
+		slot_l_hand_str = 'icons/mob/inhands/weapons/melee_left.dmi',
+		slot_r_hand_str = 'icons/mob/inhands/weapons/melee_right.dmi',
+	)
+	icon_state = "chainsaw_off"
+	worn_icon_state = "chainsaw"
+	atom_flags = TWOHANDED
+	attack_verb = list("gored", "torn", "ripped", "shred", "slashed", "cut")
+	force = 20
+	force_wielded = 75
+	throwforce = 30
+	attack_speed = 20
+	///icon when on
+	var/icon_state_on = "chainsaw_on"
+	///sprite on the mob when off but wielded
+	var/worn_icon_state_w = "chainsaw_w"
+	///sprite on the mob when on
+	var/worn_icon_state_on = "chainsaw_on"
+	///amount of fuel stored inside
+	var/max_fuel = 50
+	///amount of fuel used per hit
+	var/fuel_used = 5
+	///additional damage when weapon is active
+	var/additional_damage = 75
+
+/obj/item/weapon/twohanded/chainsaw/Initialize(mapload)
+	. = ..()
+	create_reagents(max_fuel, null, list(/datum/reagent/fuel = max_fuel))
+	AddElement(/datum/element/strappable) //Chainsaw can be strapped on hand
+
+///handle icon change
+/obj/item/weapon/twohanded/chainsaw/update_icon_state()
+	. = ..()
+	if(active) //weapon is active
+		icon_state = icon_state_on
+		return
+	else
+		icon_state = initial(icon_state)
+
+///handle worn_icon change
+/obj/item/weapon/twohanded/chainsaw/update_item_state(mob/user)
+	. = ..()
+	if(active) //weapon is active
+		worn_icon_state = worn_icon_state_on
+		return
+	if(CHECK_BITFIELD(item_flags, WIELDED)) // weapon is wielded but off
+		worn_icon_state = worn_icon_state_w
+		return
+	else
+		worn_icon_state = initial(worn_icon_state)
+
+///proc to turn the chainsaw on or off
+/obj/item/weapon/twohanded/chainsaw/proc/toggle_motor(mob/user)
+	if(active && reagents.get_reagent_amount(/datum/reagent/fuel) >= fuel_used) //check if theres enough fuel to activate)
+		force += additional_damage
+		playsound(loc, 'sound/weapons/chainsawhit.ogg', 100, 1)
+		hitsound = 'sound/weapons/chainsawhit.ogg'
+		to_chat(user, span_warning("\The [src]'s motor whirr to lifel!"))
+		update_icon()
+		update_item_state()
+	else
+		force = initial(force)
+		hitsound = initial(hitsound)
+		if(reagents.get_reagent_amount(/datum/reagent/fuel) < fuel_used)
+			to_chat(user, span_warning("\The [src] doesn't have enough fuel!"))
+			return
+		to_chat(user, span_warning("\The [src]'s motor died down!"))
+		update_icon()
+		update_item_state()
```

Comment:
early returns, use balloons pls

Reply:
I swear something went wrong here when i tried using early returns then i went back to this, ill tinker with it again for a bit. (Pain of 1 month gap between coding and review)